tests/callbacks/test_Events.cpp

```diff
@@ -0,0 +1,252 @@
+};
+
+template <DataEvent EventType>
+struct EventTest<EventType> : public ::testing::Test
+{
+    EventType event {.alloc = {.kpsh = Kokkos::Profiling::make_space_handle("Host"), .name = "my data event", .size = 2}};
+    std::string expt_descr {"{name = \"my data event\", space = \"Host\", ptr = 0, size = 2}"};
+};
+
+template <>
+struct EventTest<BeginDeepCopyEvent> : public ::testing::Test
+{
+    BeginDeepCopyEvent event {
+        .dst = {.kpsh = Kokkos::Profiling::make_space_handle("Cuda"), .name = "my destination", .size = 2},
+        .src = {.kpsh = Kokkos::Profiling::make_space_handle("Host"), .name = "my source",      .size = 2}
+    };
+    std::string expt_descr {"{src = \"my source\" (Host, 0) -> dst = \"my destination\" (Cuda, 0) of size 2}"};
+};
+
+template <>
+struct EventTest<EndDeepCopyEvent> : public ::testing::Test
+{
+    EndDeepCopyEvent event {};
+    std::string expt_descr {"{}"};
+};
+
+template <>
+struct EventTest<CreateProfileSectionEvent> : public ::testing::Test
+{
+    CreateProfileSectionEvent event {.name = "my profile section", .section_id = 1};
+    std::string expt_descr {"{name = \"my profile section\", section_id = 1}"};
+};
+
+template <ProfileSectionManipulationEvent EventType>
+struct EventTest<EventType> : public ::testing::Test
+{
+    EventType event {.section_id = 1};
+    std::string expt_descr {"{section_id = 1}"};
+};
+
+template <>
+struct EventTest<PushRegionEvent> : public ::testing::Test
+{
+    PushRegionEvent event {.name = "my push region event"};
+    std::string expt_descr {"{name = \"my push region event\"}"};
+};
+
+template <>
+struct EventTest<PopRegionEvent> : public ::testing::Test
+{
+    PopRegionEvent event {};
+    std::string expt_descr {"{}"};
+};
+
+template <>
+struct EventTest<ProfileEvent> : public ::testing::Test
+{
+    ProfileEvent event {.name = "my profile event"};
+    std::string expt_descr {"{name = \"my profile event\"}"};
+};
+
+template <typename>
+struct ImplEventTestTypes;
+
+template <typename... EventTypes>
+struct ImplEventTestTypes<Kokkos::Impl::type_list<EventTypes...>>
+{
+    using type = ::testing::Types<EventTypes...>;
+};
+
+using EventTestTypes = typename ImplEventTestTypes<EventTypeList>::type;
+
+TYPED_TEST_SUITE(EventTest, EventTestTypes);
+
+//! @test Check that @ref Kokkos::utils::callbacks::Event appropriately constrains all the event types.
+TYPED_TEST(EventTest, concept)
+{
+    static_assert(Event<TypeParam>);
+}
+
+//! @test Check that each event type is an aggregate, default constructible, and movable.
+TYPED_TEST(EventTest, traits)
+{
+    static_assert(std::is_aggregate_v<TypeParam>);
+    static_assert(std::is_default_constructible_v<TypeParam>);
+    static_assert(std::movable<TypeParam>);
+}
+
+//! @test Check that the event description is as expected.
+TYPED_TEST(EventTest, description)
+{
+    ASSERT_THAT(this->event, WhenInsertedIntoStream(::testing::StrEq(get_name<TypeParam>() + ": " + this->expt_descr)));
```

Comment:
Much easier to read (not tested):
```suggestion
    ASSERT_THAT([]{ std::stringstream oss; oss << this->event; return oss.str(); }(), ::testing::StrEq(get_name<TypeParam>() + ": " + this->expt_descr));
```

Reply:
Crystal clear now! :crystal_ball: